### ISSUE TITLE
BertGCN: Transductive Text Classification by Combining GCN and BERT

### ISSUE THREAD

**Issue by SOOJEONGKIMM** [SOOJEONGKIMM/Paper_log#10]
https://aclanthology.org/2021.findings-acl.126.pdf
ACL-IJCNLP 2021

**Comment by SOOJEONGKIMM** [comment]
![image](https://user-images.githubusercontent.com/48557539/182786269-16a9748f-f581-4805-a1af-012f94945705.png)

**Comment by SOOJEONGKIMM** [comment]
Introduction
Transductive Learning(연역법)_Text Classification Task, GNN approach
1. GNNs and Transductive Learning does not depend decisions merely on itself, but also its neighbors.
2. at training time, model influence from supervised labels through edges. Unlabeled data also contributes. consequently higher performance.

**Comment by SOOJEONGKIMM** [comment]
BertGCN model successfully combines the powers of large-scale pretraining and graph networks. 
on wide range of text classification datasets.

**Comment by SOOJEONGKIMM** [comment]
Method
1. BertGCN:
graph construct (word nodes and document nodes) following TextGCN(2019) 
define word-document edges and word-word edges based on TF-IDF and PPMI
![image](https://user-images.githubusercontent.com/48557539/182794685-e9dd88e2-56e3-4457-a4aa-a9628ed2b73a.png)

*TextGCN:
![image](https://user-images.githubusercontent.com/48557539/182795484-d99aa0a8-5d7f-434e-aa3a-5564a94e30f9.png)
initial node features
n_doc: document node, n_word: word node, d: embedding dimension


![image](https://user-images.githubusercontent.com/48557539/182795967-45d024a8-cc95-4978-aaa1-cd30e306810d.png)
ith GCN layer's output feature. p: activation function, A~: normalized adjacency matrix, W: weight matrix of the layer. L: input feature matrix of model.


![image](https://user-images.githubusercontent.com/48557539/182796551-f072a525-e90c-4c2b-ab37-30941535eb7c.png)
fed to softmax layer for classification. 
optimize parameters by cross entropy loss over labeled document nodes.

**Comment by SOOJEONGKIMM** [comment]
Interpolating BERT and GCN Predictions:
![image](https://user-images.githubusercontent.com/48557539/182796981-0c449154-e068-4ffe-9ab4-996294e5a111.png)
![image](https://user-images.githubusercontent.com/48557539/182797011-d7379171-e034-4f17-8024-07b7d6e0d444.png)
lambda parameter: tradeoff between two objectives. =>overcome drawbacks such as gradient vanishing or over-smoothing.

**Comment by SOOJEONGKIMM** [comment]
Optimization using Memory Bank

**Comment by SOOJEONGKIMM** [comment]
Experiments:
Experiments Setups
![image](https://user-images.githubusercontent.com/48557539/182798584-2e5d1548-5a4f-47f7-85e5-734c6b2c3206.png)


Main Results:
![image](https://user-images.githubusercontent.com/48557539/182798677-69fac8aa-eada-4c4d-9708-10f651562577.png)

**Comment by SOOJEONGKIMM** [comment]
The Effect of lambda:
lambda controls the trade-off between training BertGCN and BERT.
model reaches best when lambda=0.7

**Comment by SOOJEONGKIMM** [comment]
The Effect of Strategies in Joint Training:
![image](https://user-images.githubusercontent.com/48557539/182799144-83a019ca-eb9a-433c-8b31-f23544b35418.png)
result with dataset 20ng

BertGCN strategies
1. additional using fine-tuning (=RoBERTaGCN)
2. smaller learning rate
